tests/flytekit/common/workflows/sagemaker.py

```diff
@@ -0,0 +1,203 @@
+        "MaxWaitTimeInSeconds": 43200
+    },
+    algorithm_specification={
+        "TrainingInputMode": "File",
+        "AlgorithmName": "xgboost",
+        "Version": "0.72",
+    },
+    cache_version='1',
+    cachable=True,
+)
+
+simple_hpojob_task = SdkSimpleHPOJobTask(
+    max_number_of_training_jobs=10,
+    max_parallel_training_jobs=5,
+    cache_version='1',
+    retries=2,
+    cachable=True,
+)
+
+
+def calculate_hyperparameter_max_range(a, b):
+    return a + b
+
+def sample_eval_function(y_predicted, y_true):
+    assert(y_predicted.shape[0] > 0)
+    assert(y_predicted.shape == y_true.shape)
+    err_count = 0
+    for i in enumerate(y_predicted):
+        if y_predicted[i] != y_true[i]:
+            err_count += 1
+
+    return "err rate", err_count/len(y_predicted)
+
+
+@inputs(
+    extra_input1=Types.Integer,
+    extra_input2=Types.Integer,
+)
+@outputs(
+    extra_output1=Types.Blob,
+)
+@trainingjob_task(
+    trainingjob_conf={
+        "InstanceType": "ml.m4.xlarge",
+        "InstanceCount": 1,
+        "VolumeSizeInGB": 25,
+    },
+    stopping_condition={
+        "MaxRuntimeInSeconds": 43200,
+        "MaxWaitTimeInSeconds": 43200
+    },
+    algorithm_specification={
+        "TrainingInputMode": "File",
+        "AlgorithmName": "xgboost",
+        "Version": "0.72",
+    },
+    cache_version='1',
+    retries=2,
+    cachable=True
+)
+def custom_trainingjob_task(
+        wf_params,
+        train,
+        validation,
+        static_hyperparameters,
+        extra_input1,
+        extra_input2,
+        model,
+        extra_output1,
+    ):
+
+    with train as reader:
+        train_df = reader.read(concat=True)
+        dtrain_x = xgb.DMatrix(train_df[:-1])
+        dtrain_y = xgb.DMatrix(train_df[-1])
+    with validation as reader:
+        validation_df = reader.read(concat=True)
+        dvalidation_x = xgb.DMatrix(validation_df[:-1])
+        dvalidation_y = xgb.DMatrix(validation_df[-1])
+
+    static_hyperparameters["num_round"] = extra_input1 + extra_input2
+
+    my_model = xgb.XGBModel(**static_hyperparameters)
+
+    my_model.fit(dtrain_x,
+                 dtrain_y,
+                 eval_set=[(dvalidation_x, dvalidation_y)],
+                 eval_metric=sample_eval_function)
+
+    model.set(my_model)
+    extra_output1.set(my_model.evals_result())
+
+
+@workflow_class
+class SageMakerSimpleWorkflow(object):
+    a = Input(Types.Integer, required=True, help="Test integer with no default")
+    b = Input(Types.Integer, required=False, default="1", help="Test integer with default")
+
+    my_simple_xgboost_trainingjob_task = simple_xgboost_trainingjob_task(
+        train='s3://my-bucket/training.csv',
+        validation='s3://my-bucket/validation.csv',
+        static_hyperparameters=example_hyperparams,
+    )
+
+    my_simple_hpojob_task = simple_hpojob_task(
+        trainingjob_task=my_simple_xgboost_trainingjob_task,  # TODO: this doesn't seem totally reasonable
```

Comment:
@kumare3 I don't think this works actually... but I haven't come up with a intuitive way to do this